packages/sanity/src/core/releases/tool/detail/ReleaseDetail.tsx

```diff
@@ -66,6 +81,96 @@ export const ReleaseDetail = () => {
       _searchParams: [],
     })
   }, [router])
+
+  const publishTooltipContent = useMemo(() => {
+    if (!hasDocumentValidationErrors && !isValidatingDocuments) return null
+
+    const tooltipText = () => {
+      if (isValidatingDocuments) {
+        return 'Validating documents...'
+      }
+
+      if (hasDocumentValidationErrors) {
+        return 'Some documents have validation errors'
+      }
+
+      return null
+    }
+
+    return (
+      <Flex gap={1} align="center">
+        <ErrorOutlineIcon />
+        <Text muted size={1}>
+          {tooltipText()}
+        </Text>
+      </Flex>
+    )
+  }, [hasDocumentValidationErrors, isValidatingDocuments])
+
+  // review screen will not be available once published
+  // so redirect to summary screen
+  useEffect(() => {
+    if (activeScreen === 'review' && bundle?.publishedAt) {
+      navigateToSummary()
+    }
+  }, [activeScreen, bundle?.publishedAt, navigateToSummary])
+
+  const handleConfirmPublishAll = useCallback(async () => {
```

Comment:
suggestion: could we move this, and the specific logic related to the publish to a new file? And then import in this file something like:
`import {ReleasePublishAction} from "./ReleasePublishAction"` 

I think it will make it easier for our future selfs.

Reply:
Sure thing

- [ ] Extract publish all logic into separate component